packages/x-charts/src/ScatterChart/Scatter.tsx

```diff
@@ -132,29 +140,117 @@ function Scatter(props: ScatterProps) {
                 dataIndex: dataPoint.dataIndex,
               }))
           }
-          cursor={onItemClick ? 'pointer' : 'unset'}
-          {...dataPoint.interactionProps}
+          interactionProps={dataPoint.interactionProps}
+          slots={slots}
+          slotProps={slotProps}
         />
       ))}
     </g>
   );
 }
 
+export interface ScatterMarkerElementProps extends ScatterMarkerOwnerState {
+  slots?: ScatterMarkerSlots;
+  slotProps?: ScatterMarkerSlotProps;
+  onItemClick: ScatterMarkerProps['onClick'];
+  interactionProps: ReturnType<ReturnType<typeof useInteractionItemProps>>;
+}
+
+export interface ScatterMarkerSlots {
+  /**
+   * The component that renders the marker for a scatter point.
+   * @default ScatterMarker
+   */
+  marker?: React.JSXElementConstructor<ScatterMarkerProps>;
+}
+
+export interface ScatterMarkerSlotProps {
+  marker?: ScatterMarkerProps;
+}
```

Comment:
We have been trying to move `slots/slotProps` and helper types into a file ending with `.types.ts` so we are more clear about what is in there.

For some slots it might be interesting to also provide a `MyComponentSlotExtension` like `ChartsLegendSlotExtension` which already types the `slot/slotProps` in a single type for reuse.

Reply:
> We have been trying to move slots/slotProps and helper types into a file ending with .types.ts so we are more clear about what is in there.

Sure 👍 

> For some slots it might be interesting to also provide a MyComponentSlotExtension like ChartsLegendSlotExtension which already types the slot/slotProps in a single type for reuse.

Do you think it would make sense to do that in this specific case? I'm not sure where I could use it in this context.

By the way, I tried using `satisfies ScatterMarkerOwnerState` but it broke `react-docgen` 😅

Reply:
> Do you think it would make sense to do that in this specific case? I'm not sure where I could use it in this context.

I'm not sure either. It was just a point for you to keep in mind. When the API get more flushed out you can think of these details

Reply:
> > Do you think it would make sense to do that in this specific case? I'm not sure where I could use it in this context.
> 
> I'm not sure either. It was just a point for you to keep in mind. When the API get more flushed out you can think of these details

Makes sense, thanks 😄